# Dev configuration
version: '3.1'
services:
  selenium:
    image: selenium/standalone-chrome
    links:
      - nginx
    networks:
      internal:
    expose:
      - "4444"

  protractor:
    build:
      context: .
      dockerfile: Dockerfile.protractor
    image: mapstory/protractor
    links:
      - selenium
      - nginx
    command: /bin/true
    volumes:
      - ./mapstory/tests/errors:/opt/mapstory/tests/e2e/errors
    networks:
      internal:

  postgres:
    image: mdillon/postgis:9.5
    env_file:
      - docker/env/dev/db_auth.env
      - docker/env/dev/postgres.env
    volumes:
      - postgres:/var/lib/postgresql/data
    networks:
      internal:
    expose:
      - "5432"

  pgadmin:
    command: --init-db
    links:
      - postgres
    env_file:
      - docker/env/dev/db_auth.env

  elasticsearch:
    ports:
      - "9200:9200"

  rabbitmq:
    environment:
      - RABBITMQ_DEFAULT_USER=mapstory
      - RABBITMQ_DEFAULT_PASS=password
      - RABBITMQ_DEFAULT_VHOST=mapstory

  django:
    links:
      - postgres
    env_file:
      - docker/env/dev/public_host.env
      - docker/env/dev/geoserver.env
      - docker/env/dev/mapstory.env
      - docker/env/dev/db_auth.env
    command: --init-db --test
    volumes:
      - ./cover:/srv/mapstory/cover
      - ./.git:/srv/mapstory/.git

  celery:
    links:
      - postgres
    env_file:
      - docker/env/dev/public_host.env
      - docker/env/dev/geoserver.env
      - docker/env/dev/mapstory.env
      - docker/env/dev/db_auth.env
    environment:
      CELERY_NUM_WORKERS: 1

  geoserver:
    links:
      - postgres
    env_file:
      - docker/env/dev/public_host.env
      - docker/env/dev/geoserver.env
      - docker/env/dev/db_auth.env

  nginx:
    env_file:
      - docker/env/dev/public_host.env
    networks:
      internal:
        aliases:
          - docker

volumes:
  postgres:
